src/types/message-store.ts

```diff
@@ -35,7 +35,13 @@ export interface MessageStore {
   /**
    * Queries the underlying store for messages that match the provided filter.
    */
-  query(tenant: string, filter: Filter, options?: MessageStoreOptions ): Promise<GenericMessage[]>;
+  query(
+    tenant: string,
+    filter: Filter,
+    dateSort: MessageSort,
+    pagination: Pagination,
```

Comment:
Design: it seems to be a bit of anti-pattern to require a pagination object, then to have all its properties optional: we might as well make the `pagination` optional if that's the case to make calls cleaner.

Same comment for `dataSort`.

Seeing how many lines need to change by making `dataSort` and `pagination` required, I am willing to bend my default stance of making them required, however it does mean that "higher-level" logic (specifically a default for `dataSort`) will have to bleed into implementation. This is a point of discussion I suppose!

Reply:
I agree actually, I wanted to make it required due to a prior suggestion of yours, but it did feel off going through and adding empty objects everywhere.

I'll go ahead and change this to optional on the interface and where it's been modified.

I do think further down, specifically when this isn't just sugar, we will want to go in to those specific places and change some of those calls to be more specific (many of them really just want 1 item, sorted a specific way).